python/tank/descriptor/descriptor_installed_config.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+from __future__ import with_statement
+
+import os
+
+from .descriptor_config_base import ConfigDescriptorBase
+from .. import pipelineconfig_utils
+from .. import LogManager
+from ..util import ShotgunPath
+
+from ..errors import TankNotPipelineConfigurationError, TankFileDoesNotExistError, TankInvalidCoreLocationError
+
+log = LogManager.get_logger(__name__)
+
+
+class InstalledConfigDescriptor(ConfigDescriptorBase):
+    """
+    Descriptor that describes an installed Toolkit Configuration.
```

Comment:
I wonder if instead I shouldn't refactor the code so that ConfigDescriptor is the base class and is the one that is documented. Then we have two derived classes, InstalledConfigDescript and CachedConfigDescriptor (or whatever name you prefer), which are not documented, because the base class abstracts out both behaviours. After all the interface is the same for both derived classes so there's no need to document it any further.

Then we can update the io descriptor section to document the new `installed_config` descriptor.